.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "abc_submodule"]
+	path = abc_submodule
```

Comment:
I don't really like this. If someone downloads a `.tar.gz` from the GitHub archive, the `abc` folder for the submodule is going to be just empty. There's no need for them to have different names.

Reply:
The reasoning behind that is that we want to support both submodule and downloading a tar.gz, and abc should always be located in abc. So I create a symbolic link from abc_submodule to abc.

If we locate the submodule in abc, using a tar.gz would be more difficult and messy, I think.

Also, abc_submodule is a terrible name, suggestions welcome.

Reply:
> If we locate the submodule in abc, using a tar.gz would be more difficult and messy, I think.

This is *solely* relevant for the case where you download a git archive from github, where the submodule is empty. So the abc folder is free to use then.

Reply:
So would you recommend changing the submodule path?

Reply:
Yeah, put it in `abc` unless there's some other reason that hasn't been brought up yet.

Reply:
But if the submodule is located in abc, abc will not be empty, and extracting the tar.gz into abc would not make much sense. Am I missing something?

Reply:
Yes, you are. If you download the .tar.gz archive from GitHub, no submodules are included, and the directory is empty. (If GitHub included submodules in the .tar.gz download none of this would've been necessary in the first place.)

Reply:
Yeah. I think the only thing you actually need to watch out for is that there isn't necessarily a valid git repository under `yosys/abc`.